src/Frank/OpenApi.fs

```diff
@@ -0,0 +1,62 @@
+module Frank.OpenApi
+
+open System.Collections.Generic
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.Http
+open Microsoft.AspNetCore.Routing
+open Microsoft.AspNetCore.Routing.Template
+open Microsoft.Extensions.DependencyInjection
+open Microsoft.OpenApi
+open Microsoft.OpenApi.Extensions
+open Microsoft.OpenApi.Models
+
+module OperationType =
+
+    let ofHttpMethod httpMethod =
+        if HttpMethods.IsGet httpMethod then OperationType.Get
+        elif HttpMethods.IsPut httpMethod then OperationType.Put
+        elif HttpMethods.IsPost httpMethod then OperationType.Post
+        elif HttpMethods.IsDelete httpMethod then OperationType.Delete
+        elif HttpMethods.IsOptions httpMethod then OperationType.Options
+        elif HttpMethods.IsHead httpMethod then OperationType.Head
+        elif HttpMethods.IsPatch httpMethod then OperationType.Patch
+        elif HttpMethods.IsTrace httpMethod then OperationType.Trace
+        else failwithf "Unrecognized Open API HTTP method %s." httpMethod
+
+module InlineConstraint =
+    let tryParse (inlineConstraints:seq<InlineConstraint>) =
+        match Seq.tryHead inlineConstraints with
+        | Some ic ->
+            match ic.Constraint with
+            | "int32" | "long64" | "int" | "long" -> "integer"
+            | "single" | "decimal" | "double" | "float" -> "number"
+            | "boolean" | "bool" -> "boolean"
+            | "datetime" -> "string"
+            | _ -> "string"
+        | None -> "string"
+
+let emptyDocument (title, version) =
+    OpenApiDocument(
+        Info = OpenApiInfo(Title=title, Version=version),
```

Comment:
Title and Version are required according to the specification, so they are required inputs when creating a document.